src/compiler/checker.ts

```diff
@@ -21033,53 +21151,94 @@ namespace ts {
                             }
-                            if (flags & ModifierFlags.Private) {
+                            else if (flags & ModifierFlags.Private) {
                                 return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "private", "abstract");
                             }
                         }
 
                         flags |= ModifierFlags.Abstract;
                         break;
 
-                    case SyntaxKind.AsyncKeyword:
-                        if (flags & ModifierFlags.Async) {
-                            return grammarErrorOnNode(modifier, Diagnostics._0_modifier_already_seen, "async");
-                        }
-                        else if (flags & ModifierFlags.Ambient || isInAmbientContext(node.parent)) {
-                            return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_in_an_ambient_context, "async");
-                        }
-                        else if (node.kind === SyntaxKind.Parameter) {
-                            return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_appear_on_a_parameter, "async");
-                        }
-                        flags |= ModifierFlags.Async;
-                        lastAsync = modifier;
-                        break;
+                   case SyntaxKind.OverrideKeyword:
+                       if (flags & ModifierFlags.Override) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_already_seen, "override");
+                       }
+                       else if (flags & ModifierFlags.Static) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_must_precede_1_modifier, "override", "static");
+                       }
+                       else if (flags & ModifierFlags.Async) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_must_precede_1_modifier, "override", "async");
+                       }
+                       else if (flags & ModifierFlags.Readonly) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_must_precede_1_modifier, "override", "readonly");
+                       }
+                       else if (flags & ModifierFlags.Abstract) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "abstract", "override");
+                       }
+                       else if (flags & ModifierFlags.Private) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "private", "override");
+                       }
+                       else if (node.kind === SyntaxKind.Parameter) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_appear_on_a_parameter, "override");

```

Comment:
Actually, I think we should allow `override` as a marker of a parameter property. E.g.:
```ts
abstract class A {
	abstract x: number;
}
class B extends A {
	constructor(override x: number)
}
```

Reply:
Agreed

Reply:
Done